src/main/java/Course.java

```diff
@@ -0,0 +1,27 @@
+import java.util.ArrayList;
+
+// Course class representing a course, with fields courseCode, title, credits,
+// department and teacher
+public class Course {
+    public Course(String courseCode, String name, int credits, Teacher teacher) {
+        this.courseCode = courseCode;
+        this.name = name;
+        this.credits = credits;
+        this.teacher = teacher;
+    }
+
+    String courseCode;
+    String name;
+    int credits;
+    Teacher teacher;
+    ArrayList<Enrollment> enrollments;
```

Comment:
If we have a separate enrollment record for a single course mapped to a single student, then we would be having a list that will be difficult to manage. Lets say if we want to find the list of students enrolled in some specific course, we would have to iterate through the whole list and keep comparing and then re-construct a new list of result.

Instead we can do something like having an Enrollment with once course pointer and a list of students for this specific course.

Reply:
done